webbeans-junit5/src/main/java/org/apache/openwebbeans/junit5/Cdi.java

```diff
@@ -80,6 +81,15 @@
      */
     boolean disableDiscovery() default false;
 
+    /**
+     * When present on a test method parameter, it will <em>not</em> be attempted to be resolved with a CDI bean.
+     */
+    @Target(PARAMETER)
```

Comment:
About qualifier point, this could be one to make the injection unresolved and therefore handled by another extension.
This also means any custom qualifier can make it so maybe we just need to document it in our doc that if an injection parameter is ambiguous/abusively taken by cdi extension, adding an unresolved qualifier solves it.
This makes your global toggle relevant.

Do you prefer this option?

Another option is to not do it in cdi extension but in another junit ext @CdiParameterInjections which uses CDI to impl it. This way both are composable.
We can even do a junit5 stereotype combining both @CdiWithParameterInjections.

Sounds more modulable this way and avoids an useless coupling.

Wdyt?

(Still thinking to the smoothest option).

Reply:
As I imagine two extensions at first glance, the existing Cdi and a new CdiParameterInjections one, these would be heavily coupled in order to respect and comply with all the cdi definitions. Or how could for example injecting the same @ApplicationScoped bean instance be achieved without sharing the same container or any other (too) tight coupling?

To the first point with the global toggle, I'm still having difficulty to understand why it's necessary in the first place. As long as it does no harm, fine, and I don't generally object it. But it would open the point about `extensionContext.getParent()` again we already encountered earlier. At the moment I would only see the option to scan through all super and enclosing classes' `@Cdi` annotations which might not even always find a unique result (and potentially resolving as well

> // todo: enhance the setup to be thread safe? see Meecrowave ClassLoaderLock class and friends

on top of CdiExtension along the way).

Reply:
Both extension are transversal, you can use one without the other (both ways).
Typically the parameter enricher extension can be used in tomee or meecrowave too OOTB.
It just relies on a running cdi container and a CDI.current() returning the corresponding contextual CDI instance.
But ok to rework after this PR merge.